src/main/java/seedu/address/logic/util/MaximumSortedList.java

```diff
@@ -0,0 +1,189 @@
+        internalMaximumSortedList = new ArrayList<>();
+        previousMaximumSortedList.forEach(internalMaximumSortedList::add);
+    }
+
+    /**
+     * Adds the specified element to the list according to the sorted order.
+     *
+     * @param t the Comparable to be added.
+     * @return a boolean value describing whether the input element has been added.
+     */
+    public boolean add(T t) {
+        requireNonNull(t);
+        if (internalMaximumSortedList.size() >= maxSize || contains(t)) {
+            return false;
+        }
+        internalMaximumSortedList.add(t);
+        sort();
+        return true;
+    }
+
+    /**
+     * Removes the element at the specified position in this list. Shifts any
+     * subsequent elements to the left (subtracts one from their indices).
+     *
+     * @param index the index of the element to be removed.
+     * @return the Comparable at the specified index.
+     * @throws SortedListException the exception encountered when the input index
+     *         is invalid.
+     */
+    public T remove(int index) throws SortedListException {
+        if (index < 0 || index >= maxSize) {
+            throw new SortedListException("Error: The list does not contain the input index");
+        }
+        return internalMaximumSortedList.remove(index);
+    }
+
+    /**
+     * Removes the specified element in this list. Shifts any
+     * subsequent elements to the left (subtracts one from their indices).
+     *
+     * @param t the Comparable to be removed.
+     * @return the specified Comparable.
+     * @throws SortedListException the exception encountered when the list does not
+     *         contain the specified element.
+     */
+    public T remove(T t) throws SortedListException {
+        if (!contains(t)) {
+            throw new SortedListException("Error: The list does not contain the object");
+        }
+        int indexToRemove = binarySearch(t);
+        return internalMaximumSortedList.remove(indexToRemove);
+    }
+
+    /**
+     * Checks whether this list contains the specified element.
+     *
+     * @param t the Comparable to be checked against.
+     * @return a boolean value describing whether the MaximumSortedList
+     *         contains the input Comparable.
+     */
+    public boolean contains(T t) {
+        requireNonNull(t);
+        int indexOfElement = binarySearch(t);
+        if (indexOfElement == -1) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean isEmpty() {
+        return internalMaximumSortedList.isEmpty();
+    }
+
+    public void forEach(Consumer<? super T> consumer) {
+        internalMaximumSortedList.forEach(consumer);
+    }
+
+    public int size() {
+        return internalMaximumSortedList.size();
+    }
+
+    public Stream<T> stream() {
+        return internalMaximumSortedList.stream();
+    }
+
+    private int binarySearch(T t) {
```

Comment:
Nice addition of binarySearch here, so that we don't need to iterate through the whole list!